core/src/com/unciv/logic/battle/Battle.kt

```diff
@@ -226,21 +222,76 @@ object Battle {
         }
     }
 
-    private fun tryCaptureUnit(attacker: MapUnitCombatant, defender: MapUnitCombatant): Boolean {
+    private fun tryCaptureUnit(attacker: ICombatant, defender: ICombatant, attackedTile: TileInfo): Boolean {
         // https://forums.civfanatics.com/threads/prize-ships-for-land-units.650196/
-        // https://civilization.fandom.com/wiki/Module:Data/Civ5/GK/Defines
+        // https://civilization.fandom.com/wiki/Module:Data/Civ5/GK/Defines\
+        // There are 3 ways of capturing a unit, we separate them for cleaner code but we also need to ensure a unit isn't captured twice
+
+        if (defender !is MapUnitCombatant || attacker !is MapUnitCombatant) return false
+
+        if (!defender.isDefeated() || defender.unit.isCivilian()) return false
+
+        fun unitCapturedPrizeShipsUnique():Boolean {
+            if (attacker.unit.getMatchingUniques(UniqueType.KillUnitCapture)
+                        .none { defender.matchesCategory(it.params[0]) }
+            ) return false
+
+            val captureChance = min(0.8f, 0.1f + attacker.getAttackingStrength().toFloat() / defender.getDefendingStrength().toFloat() * 0.4f)
+            return Random().nextFloat() <= captureChance
+        }
+
+        fun unitCapturedGermanyUnique():Boolean{
+            if (!defender.getCivInfo().isBarbarian() || attackedTile.improvement != Constants.barbarianEncampment)
+                return false
+
+            var unitCaptured = false
+            // German unique - needs to be checked before we try to move to the enemy tile, since the encampment disappears after we move in
+
+            for (unique in attacker.getCivInfo().getMatchingUniques(UniqueType.GainFromEncampment)) {
+                attacker.getCivInfo().addGold(unique.params[0].toInt())
+                unitCaptured = true
+            }
+            return unitCaptured
+        }
+
+
+        fun unitCapturedOttomanUnique():Boolean{
+            var unitCaptured = false
+            for (unique in attacker.getCivInfo().getMatchingUniques(UniqueType.GainFromDefeatingUnit)) {
+                if (defender.unit.matchesFilter(unique.params[0])
+                        && attacker.isMelee()
+                ) {
+                    attacker.getCivInfo().addGold(unique.params[1].toInt())
+                    unitCaptured = true
+                }
+            }
+            return unitCaptured
+        }
 
-        if (attacker.unit.getMatchingUniques(UniqueType.KillUnitCapture)
-            .none { defender.matchesCategory(it.params[0]) }
-        ) return false
+        val wasUnitCaptured = unitCapturedPrizeShipsUnique() || unitCapturedGermanyUnique() || unitCapturedOttomanUnique()
```

Comment:
The prize ships unique merely captures the unit but the German and Ottoman uniques capture the unit and give gold, so an Ottoman privateer will first try to capture using the privateer unique before trying with the Ottoman unique. This could deprive the player of some gold if the privateer unique capture chance has a successful roll since the Ottoman unique won't have a chance to roll, but this is a gameplay choice and not strictly a code issue.

Reply:
Oh, because of the way that or operators resolve... good catch!
I could separate this into 3 lines but that's not as nice, wonder if there's a better way